astropy/stats/sigma_clipping.py

```diff
@@ -54,7 +55,11 @@ def _nanmean(array, axis=None):
     if isinstance(axis, tuple):
         array = _move_tuple_axes_first(array, axis=axis)
```

Comment:
Side note: in l.43, no need to calculate `first`, just put in `-1` in the reshape (much faster).

Reply:
@mhvk, thanks for noticing this.  I implemented it in #8418.